Use API for mass SP migration (#19)
uyuni-project/uyuni-docs: modules/upgrade/pages/client-migration.adoc

After the migration is complete, schedule a reboot of the client from the {productname} Server.

////

== SP Mass Migration

> If you want to migrate many clients to the next SP version, {suse} recommends to use {productname} API calls.

Reviewer: ```suggestion
If you want to migrate a large number of clients to the next SP version, you can use {productname} API calls.
```

> You need to know to which migration target you want to migrate.

Reviewer: I think this should probably be a procedure.

Author: ok, trying to do my best ;)

List available migration targets:
----
spacecmd api -- system.listMigrationTargets -A 1000010001
----
Create a list of system IDs you want to migrate.

> Now we can iterate over the system IDs and call [systemitem]``listMigrationTarget`` for every system and check if the wanted target is part of the available targets.

Reviewer: ```suggestion
For each system ID, call [systemitem]``listMigrationTarget`` and check that it is an available target.
```

> If yes, we call [systemitem]``system.scheduleSPMigration``, otherwise we skip the system.

Reviewer: ```suggestion
If the system ID is an available target, call [systemitem]``system.scheduleSPMigration``.
If it is not an available target, skip the system.
```

> Here is a template that you can adapt for your environment:

Reviewer: ```suggestion
Adapt this template for your environment:
```
